### PR TITLE
New select_credentials.json vector that has multiple Fields in one Input Descriptor

### LINKED ISSUE #158: Add select credentials test vector with multiple Fields
In the test vectors for the Presentation Exchange "select credentials" feature, no test case has an _Input Descriptor_ with more than one _Field_.
[test-vectors/presentation_exchange/select_credentials.json](https://github.com/TBD54566975/web5-spec/blob/main/test-vectors/presentation_exchange/select_credentials.json)

Add a new test vector that has an Input Descriptor with multiple Fields.

This is necessary to demonstrate a [bug in the web5-go implementation of SelectCredentials](https://github.com/TBD54566975/web5-go/issues/153). It might also be present in other language SDKs. 

It will also exercise the AND relationship of multiples Fields in an InputDescriptor

### PR DESCRIPTION
**Pull request by tomdaffurn**
Closes #158 

## Important!

Make sure that if you are updating a test vector that you test in the following repos before merging!

- [ ] https://github.com/TBD54566975/web5-js
- [ ] https://github.com/TBD54566975/web5-kt
- [x] https://github.com/TBD54566975/web5-go
- [ ] https://github.com/TBD54566975/web5-rs
- [ ] https://github.com/TBD54566975/web5-swift

### CODE REVIEW

**Review by decentralgabe — APPROVED** [review]

**Review by nitro-neal — APPROVED** [review]

**Comment by nitro-neal** [comment]
The updated PEX lib in js expects type to be there.

```
export interface FilterV2 {
    const?: OneOfNumberStringBoolean;
    enum?: Array<OneOfNumberStringBoolean>;
    exclusiveMinimum?: OneOfNumberString;
    exclusiveMaximum?: OneOfNumberString;
    format?: string;
    formatMaximum?: string;
    formatMinimum?: string;
    formatExclusiveMaximum?: string;
    formatExclusiveMinimum?: string;
    minLength?: number;
    maxLength?: number;
    minimum?: OneOfNumberString;
    maximum?: OneOfNumberString;
    not?: object;
    pattern?: string;
    type: string;
    contains?: FilterV2Base;
    items?: FilterV2BaseItems;
}
```

Chad said type is not mandatory for valid json schema, but highly recommend. And our pex lib implementers seem to think so as well since the only required field they put was type.

File: test-vectors/presentation_exchange/select_credentials.json [hunk 1]
```diff
@@ -265,6 +265,58 @@
...
+               "id":"whatever",
+               "purpose":"must have btcAddress, dogeAddress, and issuer",
+               "constraints":{
+                 "fields":[
+                   {
+                     "path":[
+                       "$.vc.credentialSubject.btcAddress"
+                     ],
+                      "filter":{
+                        "const": "btcAddress123"
```
Comment (nitro-neal) [inline test-vectors/presentation_exchange/select_credentials.json:287]:
```suggestion
                        "type": "string",
                        "const": "btcAddress123"
```

File: test-vectors/presentation_exchange/select_credentials.json [hunk 2]
```diff
@@ -265,6 +265,58 @@
...
+                     "path":[
+                       "$.vc.credentialSubject.dogeAddress"
+                     ]
+                   },
+                   {
+                     "path":[
+                       "$.vc.issuer"
+                     ],
+                     "filter":{
+                       "const": "did:key:z6MkfHBNmc4cChNtDvar18TZigfXt7P65FBGwqQTtxQoQPnG"
```
Comment (nitro-neal) [inline test-vectors/presentation_exchange/select_credentials.json:300]:
```suggestion
                       "type": "string",
                       "const": "did:key:z6MkfHBNmc4cChNtDvar18TZigfXt7P65FBGwqQTtxQoQPnG"
```

**Comment by nitro-neal** [comment]
this one would have to change and have type added too - https://github.com/TBD54566975/web5-spec/pull/159/files#diff-1197e3053282a8f67310a45ce5e2675a5630f3c13291c86c5a2b88928c2d6f35R249

**Comment by tomdaffurn** [comment]
I'm checking each language before I merge this. It's slow because some SDKs were using old versions of the test vectors, and do not pass with the pre-existing tests. `web5-kt` in particular @nitro-neal.

**Comment by decentralgabe** [comment]
@tomdaffurn can this be merged?

**Comment by tomdaffurn** [comment]
I think so. I was waiting to get each language SDK fixed, but I don't think we need to. They're pinned to a particular commit

**Pull request merged**